### PR TITLE
lesspipe: 1.85 -> 2.05

### PR DESCRIPTION
**Pull request by avdv**
###### Description of changes

This PR upgrades lesspipe to the latest version.

Note, this is a big upgrade in terms of (internal) file handling and brings syntax highlighting support with various tools (e.g. can use `bat`, `pygmentize`, or `source-highlight`).

Also, completion (supporting looking into archive files) should work for bash and zsh now.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by avdv** [comment]
@SuperSandro2000 can you have a look please!?

File: pkgs/tools/misc/lesspipe/default.nix [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
 
+  installFlags = [ "DESTDIR=$(out)" ];
+
+  postInstall = ''
+    for f in lesspipe.sh lesscomplete; do
+      wrapProgram "$out/bin/$f" --prefix-each PATH : "${lib.makeBinPath [ file gnused procps ]}"
+    done
+  '';
+
   src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/lesspipe/default.nix:26]: Can we please change the sorting to be more standard?

File: pkgs/tools/misc/lesspipe/default.nix [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
 stdenv.mkDerivation rec {
   pname = "lesspipe";
-  version = "1.85";
+  version = "2.05";
 
-  nativeBuildInputs = [ perl ];
+  nativeBuildInputs = [ perl makeWrapper ];
   buildInputs = [ perl bash ];
   strictDeps = true;
   preConfigure = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/lesspipe/default.nix:10]:
```suggestion
  postPatch = ''
```

File: pkgs/tools/misc/lesspipe/default.nix [hunk 3]
```diff
@@ -1,35 +1,35 @@
...
+  configurePlatforms = [ ];
   dontBuild = true;
 
+  installFlags = [ "DESTDIR=$(out)" ];
+
+  postInstall = ''
+    for f in lesspipe.sh lesscomplete; do
+      wrapProgram "$out/bin/$f" --prefix-each PATH : "${lib.makeBinPath [ file gnused procps ]}"
+    done
+  '';
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/lesspipe/default.nix:24]: Hardcoding the binaries is probably better than wrapping.

File: pkgs/tools/misc/lesspipe/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/tools/misc/lesspipe/default.nix:26]: yes

Comment (avdv) [reply pkgs/tools/misc/lesspipe/default.nix]: Done.

**Pull request merged**